src/main/scala/io/iohk/ethereum/domain/DifficultyCalculator.scala

```diff
@@ -0,0 +1,41 @@
+package io.iohk.ethereum.domain
+
+import io.iohk.ethereum.utils.BlockchainConfig
+
+class DifficultyCalculator(blockchainConfig: BlockchainConfig) {
+  val DifficultyBoundDivision: Int = 2048
+  val FrontierTimestampDiffLimit: Int = -99
+  val ExpDifficultyPeriod: Int = 100000
+  val MinimumDifficulty: BigInt = 131072
+
+  def calculateDifficulty(blockNumber: BigInt, blockTimestamp: Long, parentHeader: BlockHeader): BigInt = {
+    import blockchainConfig.{homesteadBlockNumber, difficultyBombPauseBlockNumber, difficultyBombContinueBlockNumber}
+
+    val x: BigInt = parentHeader.difficulty / DifficultyBoundDivision
+    val c: BigInt =
+      if (blockNumber < homesteadBlockNumber) {
```

Comment:
yes we do, this is used in regular sync, also there is the case of the private network